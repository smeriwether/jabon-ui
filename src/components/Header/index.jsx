import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  onSubmit: PropTypes.func.isRequired
};

const styles = {
  header: 'h-100 w-100 bg-white flex justify-between items-center pl3 pr3 bb br b--moon-gray',
  button: 'pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box'
};

const Header = ({ onSubmit }) => {
  return (
    <div className={ styles.header }>
      <div className='near-black'>
        <h2>A Modern SOAP Editor</h2>
      </div>
      <div>
        <button data-role='submit' className={ styles.button } onClick={ onSubmit }>
          Submit
        </button>
      </div>
    </div>
  );
};

Header.propTypes = propTypes;

export default Header;
