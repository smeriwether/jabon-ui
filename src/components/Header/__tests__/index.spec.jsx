import Header from '../../Header';

describe('<Header />', () => {
  const setup = (overrides) => {
    const props = {
      onSubmit: jest.fn(),
      ...overrides
    };

    const wrapper = shallow(<Header { ...props } />);

    return { wrapper };
  };

  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the onSubmit prop when the submit button is clicked', () => {
    const onSubmitMock = jest.fn();
    const { wrapper } = setup({ onSubmit: onSubmitMock });

    wrapper.find('button[data-role="submit"]').simulate('click');

    expect(onSubmitMock).toHaveBeenCalled();
  });
});
