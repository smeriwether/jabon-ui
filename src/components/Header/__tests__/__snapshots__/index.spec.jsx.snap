// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="near-black"
        >
          <h2>
            A Modern SOAP Editor
          </h2>
        </div>,
        <div>
          <button
            className="pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box"
            data-role="submit"
            onClick={[MockFunction]}
          >
            Submit
          </button>
        </div>,
      ],
      "className": "h-100 w-100 bg-white flex justify-between items-center pl3 pr3 bb br b--moon-gray",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h2>
            A Modern SOAP Editor
          </h2>,
          "className": "near-black",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A Modern SOAP Editor",
          },
          "ref": null,
          "rendered": "A Modern SOAP Editor",
          "type": "h2",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box"
            data-role="submit"
            onClick={[MockFunction]}
          >
            Submit
          </button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box",
            "data-role": "submit",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="near-black"
          >
            <h2>
              A Modern SOAP Editor
            </h2>
          </div>,
          <div>
            <button
              className="pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box"
              data-role="submit"
              onClick={[MockFunction]}
            >
              Submit
            </button>
          </div>,
        ],
        "className": "h-100 w-100 bg-white flex justify-between items-center pl3 pr3 bb br b--moon-gray",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2>
              A Modern SOAP Editor
            </h2>,
            "className": "near-black",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A Modern SOAP Editor",
            },
            "ref": null,
            "rendered": "A Modern SOAP Editor",
            "type": "h2",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box"
              data-role="submit"
              onClick={[MockFunction]}
            >
              Submit
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "pointer f6 br2 ba bg-green white pa2 pl3 pr3 ml1 mv1 hover-bg-dark-green border-box",
              "data-role": "submit",
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
